### PR TITLE
[SPARK-46091][K8S] Respect the defined container SPARK_LOCAL_DIRS env

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

In this pr, it respects the defined container SPARK_LOCAL_DIRS env, and will combine it with the spark-local-dir-* volume mount paths as the final spark local dirs.

### Why are the changes needed?

Here is a user case,

For spark on kubernetes, we leverage hostPath as spark local dirs.

But we do not mount the sub host paths directly to the pod, we mount a root path for spark driver/executor pod.

For example, the root host path is `/hadoop`.

And there are several sub disks under that, likes `hadoop/1, /hadoop/2, /hadoop/3, /hadoop4`.

And we have different machine models, some nodes have 4 disks and some nodes have 12 disks, so the sub disks numbers are different for different nodes.

So we want to define the SPARK_LOCAL_DIRS in the driver/executor pod env, the non-existing sub disks defined in `SPARK_LOCAL_DIRS` will be ignored, so that we can define the same container SPARK_LOCAL_DIRS env in the pod template for different machine models.

https://github.com/apache/spark/blob/132bb63a897f4f4049f34deefc065ed3eac6a90f/core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala#L250-L262

### Does this PR introduce _any_ user-facing change?

Yes, the user defined container SPARK_LOCAL_DIRS env will be combined into final spark local dirs.

### How was this patch tested?

UT.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala [hunk 1]
```diff
@@ -45,7 +45,7 @@ private[spark] class LocalDirsFeatureStep(
     pod.container.getEnv.asScala
       .filter(_.getName == "SPARK_LOCAL_DIRS")
       .lastOption.foreach { containerEnv =>
-      localDirs ++= containerEnv.getValue.split(",")
+      localDirs ++= Option(containerEnv.getValue).map(_.split(",")).getOrElse(Array.empty[String])
```
Comment (turboFei) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala:48]: Now env `valueFrom` is ignored here.

**Comment by turboFei** [comment]
gentle ping @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you for pinging me, @turboFei . I'll review this tonight.

**Review by dongjoon-hyun — COMMENTED** [review]
May I ask why you are unable to use the existing supported SparkConf ways? The following existing Apache Spark code handles them already.
- `SPARK_LOCAL_DIRS`
- `spark.local.dir`

https://github.com/apache/spark/blob/9987cab84bb05a61a5c0b43f94a733561a2e074a/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala#L50-L51

In addition, if you really need to override via pod environment, it seems that this PR can be replaced by the following existing workaround. Could you try to use `SPARK_EXECUTOR_DIRS` instead of your `SPARK_LOCAL_DIRS`?
```
spark.executorEnv.SPARK_EXECUTOR_DIRS
```

**Comment by turboFei** [comment]
thanks again, will close this pr.

**Comment by dongjoon-hyun** [comment]
Thank you, @turboFei !

**Pull request closed without merging**